### PR TITLE
Buttons Group

### PR DESCRIPTION
**Pull request by ludwiczakpawel**
## What type of PR is this? (check all applicable)

- [x] Feature

## Description
This PR introduces new component to Crayons library: `.crayons-btn-group` - it will let visually group 2 or buttons so they look like a one component.

This will be needed for future work around comments subscription features.

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

image

## Added tests?

- [x] no, because they aren't needed

## Added to documentation?

- [x] no documentation needed

### CODE REVIEW

File: app/assets/stylesheets/components/buttons.scss [hunk 1]
```diff
@@ -76,6 +77,12 @@
   }
 }
 
+// Icon alone
+.crayons-btn--icon-alone {
```
Comment (rhymes) [inline app/assets/stylesheets/components/buttons.scss]: `-alone`, so sad 😢

File: app/views/pages/crayons.html.erb [hunk 2]
```diff
@@ -496,8 +497,44 @@
...
-  <a href="#" class="... crayons-btn--icon-left"><svg width="24" height="24" xmlns="http://www.w3.org/2000/svg" class="crayons-icon">...</svg>Button</a>
-  <a href="#" class="... crayons-btn--secondary crayons-btn--full crayons-btn--icon-left"><svg width="24" height="24" xmlns="http://www.w3.org/2000/svg" class="crayons-icon">...</svg>Button</a>
+  <a href="#" class="... crayons-btn--icon-alone"><svg class="crayons-icon" ...>...</svg></a>
+  <a href="#" class="... crayons-btn--icon-left"><svg class="crayons-icon" ...>...</svg>Button</a>
+  <a href="#" class="... crayons-btn--secondary crayons-btn--full crayons-btn--icon-left"><svg class="crayons-icon" ...>...</svg>Button</a>
+  '.rstrip %></pre>
+
+  <div class="body">
+    <h2>Buttons Group</h2>
+    <p>It is possible to combine two or more buttons into visual group.</p>
```
Comment (rhymes) [inline app/views/pages/crayons.html.erb]: `into a visual group`

File: app/views/pages/crayons.html.erb [hunk 3]
```diff
@@ -496,8 +497,44 @@
...
+    <p>Use your best judgment for when it should be used. If you think every time you have more than one button you should create this kind of group, then no it's wrong.</p>
+    <p>Think of it as <em>an extenstion</em> to the button itself: this component lets you <strong>visually</strong> create an <em>actionable area</em> for cases when you need something more than just a classic button.</p>
```
Comment (rhymes) [inline app/views/pages/crayons.html.erb]: `extension`

File: app/assets/stylesheets/components/buttons.scss (hunk 1 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/components/buttons.scss]: hey, i could call it `.crayons-btn--icon-forever-alone` :D

**Comment by ludwiczakpawel** [comment]
@rhymes I fixed the typos, thanks for proof-reading!

> ps. why is the third row of buttons bigger than the previous two?

This style is called 'outlined' which in fact uses `border` to create that _outline_. Border itself affects the layout of any element. It affects layout even if you have `box-sizing: border-box` - since it works as long as you have defined width/height and buttons do not have it. The size of our buttons is actually dynamic: it's calculated from padding & font-size/line-height. Since outlined buttons also have border, we have to add it into calculations so when `crayons-btn-primary` is next to `crayons-btn-outlined` they have the same height - in that case we reduce vertical padding in outlined button to accommodate space for 2px outline (2px == 0.125em). That's the line: https://github.com/thepracticaldev/dev.to/blob/89208849ed1a200494f5adfa500added93f643e8/app/assets/stylesheets/components/buttons.scss#L47

Honestly, I didn't really care about left & right padding because it's unlikely we will ever have stacked buttons... BUT you made me realized it doesn't cost us anything, sooooo.... i'm fixing this in the next commit! :) Thanks!

**Comment by rhymes** [comment]
Ahah terrific explanation, thanks!

**Pull request merged**